manifests/init.pp

```diff
@@ -49,6 +50,19 @@
     $dhcp_interfaces = $interfaces
   }
 
+  # See https://tools.ietf.org/html/draft-ietf-dhc-failover-12 for why BOOTP is
+  # not supported in the failover protocol. Relay agents *can* be made to work
+  # so $bootp can be explicitly set to true to override this default.
+  if $bootp == undef {
+    if $failover {
```

Comment:
I see now that there's no dependency on stdlib in this module so let's leave that out. The inner if statement can still be simplified using pure puppet.